lib/bap/bap_self.ml

```diff
@@ -60,4 +61,94 @@ module Create() = struct
   let warning f = message Warning ~section:name f
   let error f = message Error ~section:name f
 
+  module Param = struct
+    let plugin_name = name
+
+    type 'a t = 'a ref
+    type 'a parser = string -> [ `Ok of 'a | `Error of string ]
+    type 'a printer = Format.formatter -> 'a -> unit
+    type 'a converter = 'a parser * 'a printer
+
+    let main = ref Term.(const ())
+
+    let conf_file_options : (string, string) List.Assoc.t =
+      let conf_filename =
+        let (/) = Filename.concat in
+        Bap_config.confdir / plugin_name / "config" in
+      let string_splitter str =
+        let str = String.strip str in
+        match String.split str ~on:'=' with
+        | k :: _ when String.prefix k 1 = "#" -> None
+        | [""] | [] -> None
+        | [k] -> invalid_argf
+                   "Maybe comment out \"%s\" using # in config file?" k ()
+        | k :: vs -> Some (k, String.concat ~sep:"=" vs) in
```

Comment:
`sep` should be `""`, or you can just omit it, as by default it is an empty string. Also, I think it is a good idea to `strip` both key and data.